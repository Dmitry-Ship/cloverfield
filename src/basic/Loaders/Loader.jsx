import React, { PropTypes } from 'react';

const Loader = ({type}) => (
  <div className='loader__wrapper' >
    <div className={`loader--${type}`}>
      <span className='loader__dot' />
      <span className='loader__dot' />
      <span className='loader__dot' />
    </div>
  </div>
);

export default Loader;

Loader.defaultProps = {
  type: 'spinny'
}

Loader.propTypes = {
  type: PropTypes.string
}
