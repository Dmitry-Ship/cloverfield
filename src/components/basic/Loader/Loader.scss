@value colors: "../../../styles/colors.scss";
@value accent, bg from colors;

.wrapper {
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
}

.loadingText {
  color: accent;
  text-align: center;
}

.loader {
  background-color: bg;
  margin: auto;
  width: 100px;
  height: 100px;
  border-radius: 50%;
  position: relative;
  &:before, &:after {
    width: 100px;
    height: 100px;
    border-radius: 50%;
    content: '';
    display: block;
  }
}

.spinny {
  composes: loader;
  &:after {
    position: absolute;
    top: -4px;
    left: -4px;
    border: 4px solid transparent;
    border-top-color: accent;
    border-bottom-color: accent;
    animation: spinny 1s linear infinite;
  }
}

@keyframes spinny {
  0% {
    transform: rotate(0deg) scale(1);
  }
  50% {
    transform: rotate(180deg) scale(1.5);
  }
  100% {
    transform: rotate(360deg) scale(1);
  }
}

.bouncy {
  composes: loader;
  animation: spin 2s linear infinite;
  &:before, &:after {
    height: 20px;
    width: 20px;
    background: accent;
    position: absolute;
    top: -10px;
    left: 50%;
    margin-left: -10px;
  }
  &:after {
    top: auto;
    bottom: -10px;
    animation: bouncy 0.5s ease-in-out infinite;
  }
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

@keyframes bouncy {
  0% {
    transform: translateY(0px);
  }
  50% {
    transform: translateY(30px);
  }
}

.clock {
  composes: loader;
  background: none;
  border: 4px solid #eee;
  &::after {
    width: 4px;
    height: 66px;
    background: accent;
    border-radius: 0px;
    position: absolute;
    top: 50%;
    left: 50%;
    margin-left: -2px;
    animation: spin 2s linear infinite;
    transform-origin: top center;
  }
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

.pulse {
  composes: loader;
  &:before, &:after {
    position: absolute;
    top: 50%;
    left: 50%;
    margin-left: -50px;
    margin-top: -50px;
    background: accent;
  }
  &:before {
    animation: pulse .5s linear infinite;
  }
  &:after {
    opacity: 0;
    animation: pulse .25s linear .62s infinite;
  }
}


@keyframes pulse {
  0% {
    transform: scale(0);
    opacity: 1;
  }
  100% {
    transform: scale(1.3);
    opacity: 0;
  }
}

.triple-dots {
  composes: loader;
  background: transparent;
  .dot {
    display: block;
    width: 20px;
    height: 20px;
    border-radius: 50%;
    background: accent;
    position: absolute;
    top: 50%;
    &:nth-child(1) {
      transform: translateX(-50%);
      animation: bounce .3s ease-in-out infinite;
    }
    &:nth-child(2) {
      left: 50%;
      transform: translateX(-50%);
      animation: bounce .3s ease-in-out 0.1s infinite;
    }
    &:nth-child(3) {
      left: 100%;
      transform: translateX(-50%);
      animation: bounce .3s ease-in-out 0.2s infinite;
    }
  }
}

@keyframes bounce {
  0% {
    top: 0%;
  }
  50% {
    top: 50%;
  }
  100% {
    top: 0%;
  }
}
