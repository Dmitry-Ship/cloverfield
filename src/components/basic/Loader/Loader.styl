@import '../../../styles/index.styl'

.wrapper
  abs-center()

.loader, .spinny, .bouncy, .clock, .pulse, .triple-dots
  margin auto
  circle(100px)
  background color__bg
  position relative
  &:before, &:after
    circle(100px)
    content ''
    display block

.spinny:after
  // composes loader
  absolute top -4px left -4px
  border 4px solid transparent
  border-top-color color__accent
  border-bottom-color color__accent
  animation spinny 1s linear infinite

@keyframes spinny
  0%
    transform rotate(0deg) scale(1)
  50%
    transform rotate(180deg) scale(1.2)
  100%
    transform rotate(360deg) scale(1)

.bouncy
  // composes loader
  animation spin 2s linear infinite
  &:before, &:after
    height 20px
    width 20px
    background color__accent
    absolute top -10px left 50%
    margin-left -10px
  &:after
    top auto
    bottom -10px
    animation bouncy 0.5s ease-in-out infinite

@keyframes spin
  0%
    transform rotate(0deg)
  100%
    transform rotate(360deg)
@keyframes bouncy
  0%
    transform translateY(0px)
  50%
    transform translateY(30px)


.clock
  // composes loader
  background none
  border 4px solid #eee
  &::after
    width 4px
    height 66px
    background color__accent
    border-radius 0px
    absolute top 50% left 50%
    margin-left -2px
    animation spin 2s linear infinite
    transform-origin top center

@keyframes spin
  0%
    transform rotate(0deg)
  100%
    transform rotate(360deg)


.pulse
  // composes loader
  &:before, &:after
    absolute top 50% left 50%
    margin-left -50px
    margin-top -50px
    background $color__action
  &:before
    animation pulse .5s linear infinite
  &:after
    opacity 0
    animation pulse .25s linear .62s infinite
@keyframes pulse
  0%
    transform scale(0)
    opacity 1
  100%
    transform scale(1.3)
    opacity 0

.triple-dots
  // composes loader
  height 0px
  width 0px
  position relative
  .dot
    display block
    circle(20px)
    background color__accent
    position absolute
    top 0px
    &:nth-child(1)
      left -40px
      animation bounce .3s ease-in-out infinite
    &:nth-child(2)
      animation bounce .3s ease-in-out 0.1s infinite
    &:nth-child(3)
      left 40px
      animation bounce .3s ease-in-out 0.2s infinite

@keyframes bounce
  0%
    transform translateY(0px)
  50%
    transform translateY(-30px)
