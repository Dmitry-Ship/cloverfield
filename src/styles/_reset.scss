body {
  font-family: Raleway;
}
  
a {
  text-decoration: none;
}

ul {
  list-style-type: none;
}

input, button, textarea {
  outline: none;
}

* {
  box-sizing: border-box;
}
