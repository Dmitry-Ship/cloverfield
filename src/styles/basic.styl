body
  font-family Rubik

a
  text-decoration none

ul
  list-style-type none

input, button, textarea
  outline none
*
  box-sizing border-box
