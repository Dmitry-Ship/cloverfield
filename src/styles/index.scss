@import '_normalize.scss';
@import '_basic.scss';