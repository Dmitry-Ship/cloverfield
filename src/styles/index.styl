@import 'basic.styl'
@import 'mixins.styl'
