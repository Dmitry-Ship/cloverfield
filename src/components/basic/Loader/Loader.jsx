import PropTypes from 'prop-types';
import React from 'react';
import styles from './Loader.scss';

const Loader = ({ type, className, text }) => (
  <div className={`${styles.wrapper} ${className}`} >
    <div className={styles[type]}>
      <span className={styles.dot} />
      <span className={styles.dot} />
      <span className={styles.dot} />
    </div>
    {text && <h2 className={styles.loadingText} >{text}</h2>}
  </div>
);

export default Loader;

Loader.defaultProps = {
  type: 'spinny',
  className: '',
  text: '',
};

Loader.propTypes = {
  type: PropTypes.string,
  className: PropTypes.string,
  text: PropTypes.string,
};
