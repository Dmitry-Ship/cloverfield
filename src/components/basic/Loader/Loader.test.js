import React from 'react';
import renderer from 'react-test-renderer';
import Loader from './Loader';
import styles from './Loader.scss';

describe('<Loader />', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <Loader />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render text when passed in', () => {
    const tree = renderer.create(
      <Loader text={'hello world'} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render spinny loader', () => {
    const tree = renderer.create(
      <Loader type={'spinny'} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render clock loader', () => {
    const tree = renderer.create(
      <Loader type={'clock'} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render bouncy loader', () => {
    const tree = renderer.create(
      <Loader type={'bouncy'} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render triple-dots loader', () => {
    const tree = renderer.create(
      <Loader type={'triple-dots'} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
