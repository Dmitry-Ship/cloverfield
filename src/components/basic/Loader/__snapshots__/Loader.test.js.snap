// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should render bouncy loader 1`] = `
<div
  className="wrapper "
>
  <div
    className="bouncy"
  >
    <span
      className="dot"
    />
    <span
      className="dot"
    />
    <span
      className="dot"
    />
  </div>
  
</div>
`;

exports[`<Loader /> should render clock loader 1`] = `
<div
  className="wrapper "
>
  <div
    className="clock"
  >
    <span
      className="dot"
    />
    <span
      className="dot"
    />
    <span
      className="dot"
    />
  </div>
  
</div>
`;

exports[`<Loader /> should render correctly 1`] = `
<div
  className="wrapper "
>
  <div
    className="spinny"
  >
    <span
      className="dot"
    />
    <span
      className="dot"
    />
    <span
      className="dot"
    />
  </div>
  
</div>
`;

exports[`<Loader /> should render spinny loader 1`] = `
<div
  className="wrapper "
>
  <div
    className="spinny"
  >
    <span
      className="dot"
    />
    <span
      className="dot"
    />
    <span
      className="dot"
    />
  </div>
  
</div>
`;

exports[`<Loader /> should render text when passed in 1`] = `
<div
  className="wrapper "
>
  <div
    className="spinny"
  >
    <span
      className="dot"
    />
    <span
      className="dot"
    />
    <span
      className="dot"
    />
  </div>
  <h2
    className="loadingText"
  >
    hello world
  </h2>
</div>
`;

exports[`<Loader /> should render triple-dots loader 1`] = `
<div
  className="wrapper "
>
  <div
    className="triple-dots"
  >
    <span
      className="dot"
    />
    <span
      className="dot"
    />
    <span
      className="dot"
    />
  </div>
  
</div>
`;
